corehq/apps/userreports/views.py

```diff
@@ -1204,12 +1204,16 @@ def post(self, request, *args, **kwargs):
         try:
             if self.edit_form.is_valid():
                 config = self.edit_form.save(commit=True)
-                messages.success(request, _('Data source "{}" saved!').format(
+                message_list = [_('Data source "{}" saved.').format(
                     config.display_name
-                ))
-                messages.success(request, _(
-                    'This data source will be built / rebuilt automatically by CommCare HQ.'
-                ))
+                )]
+                if config.meta.build.awaiting:
+                    message_list.append(_(
+                        'This data source will be built / rebuilt automatically by CommCare HQ.'
+                    ))
+                else:
+                    message_list.append(_('This change does not require a rebuild.'))
```

Comment:
nit: Can we either skip it or say that "HQ will not perform a data source rebuild for the changes made. Please initiate one manually if needed."

It appears that a rebuild might be needed at times, for instance, adding a nullable column is not considered to be rebuildable, so the user would need to rebuild separately if they want to populate existing rows
https://github.com/dimagi/commcare-hq/blob/0b590d2e093fe694e54d8a8ebdca0c79246c97c5/corehq/apps/userreports/alembic_diffs.py#L113

Reply:
Good call. 7253e9aef29